bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/WorkbookEditorsHandler.java

```diff
@@ -98,6 +114,159 @@ private List<EditorReference> getParts(WorkbenchPage page) {
 		return refs;
 	}
 
+	/**
+	 * Generates a mapping of EditorReferences to display label texts. If display
+	 * names collide parent directories will be added until the EditorReference can
+	 * be differentiated from all references that share the same file name. If the
+	 * collisions share multiple paths segments only the first differing path
+	 * segment will be prepended.<br>
+	 * <br>
+	 * Example where all collisions share the same segments:
+	 *
+	 * <pre>
+	 * /project/test1/foo/bar/file -> test1/file
```

Comment:
we would expect `test1/.../file` here (or even `file (in test1/...)`, silently skipping intermediary segments can be interpreted as a bug.

Reply:
I've implemented the  `test1/.../file` variant, the user can see the full path in the tooltip.

![Selection_222](https://user-images.githubusercontent.com/459631/189994014-d39633f6-ff6a-4f0e-8899-3b0ad1f1ef54.png)

Reply:
I've also thought about styling the marker for the omitted part differently, maybe grey it out, but then it gets confusing for the user with regards to the filter behavior.
Should the filter match on the greyed out `/.../` or should the user be able to match on `test1/file`, ignoring the `/.../` part?
Maybe it's better to keep the styling as it is. :shrug:

Reply:
IMO, current/default styling is good enough.